File: backend/app/models/approval.py
from pydantic import BaseModel
from typing import Optional

class Approval(BaseModel):
    id: str
    session_id: str
    function_call_id: str
    description: str
    status: str = "pending"  # pending, approved, rejected
    created_at: int
    updated_at: int
    result: Optional[str] = None

class ApprovalRequest(BaseModel):
    session_id: str
    function_call_id: str
    description: str

class ApprovalResponse(BaseModel):
    id: str
    status: str
    result: Optional[str] = None

File: frontend/src/types/approval.ts
export interface Approval {
  id: string;
  session_id: string;
  function_call_id: string;
  description: string;
  status: 'pending' | 'approved' | 'rejected';
  created_at: number;
  updated_at: number;
}

export interface ApprovalState {
  approvals: Approval[];
  loading: boolean;
  error: string | null;
}

export interface ApprovalResponse {
  success: boolean;
  message: string;
  approval?: Approval;
}
